spec/services/sessions/retrieve_spec.rb

```diff
@@ -0,0 +1,112 @@
+require 'rails_helper'
+
+describe Sessions::Retrieve do
+  describe 'process' do
+    before(:each) do
+      @user1 = FactoryGirl.create(:david)
+      @user2 = FactoryGirl.create(:omar)
+
+      @active_user1_token = Sessions::GenerateToken.new(@user1.id, 2.seconds.ago).process
+      @active_user2_token = Sessions::GenerateToken.new(@user2.id, 2.seconds.ago).process
+
+      @active_user1_session = Session.create! user: @user1, token: @active_user1_token
+      @active_user2_session = Session.create! user: @user2, token: @active_user2_token
+
+      @inactive_user1_token = Sessions::GenerateToken.new(@user1.id, 2.hours.ago).process
+      @inactive_user2_token = Sessions::GenerateToken.new(@user2.id, 2.hours.ago).process
+
+      @inactive_user1_session = Session.create! user: @user1, token: @inactive_user1_token, active: false
+      @inactive_user2_session = Session.create! user: @user2, token: @inactive_user2_token, active: false
+    end
```

Comment:
In testing you don't need to be DRY, actually the more the test can tell you, the better, so probably you can go back to the way the tests were before the refactoring. The goal is to make tests more self-explanatory.